src/components/elements/Menu/MenuCollapse.tsx

```diff
@@ -0,0 +1,292 @@
+import classNames from "classnames";
+import { ReactNode, useEffect, useRef, useState } from "react";
+import React from "react";
+import { Else, If, Then } from "react-if";
+import { twMerge as tw } from "tailwind-merge";
+
+import Icon, { IconNames } from "@/components/extensive/Icon/Icon";
+
+import { MenuItem } from "../MenuItem/MenuItem";
+import { MENU_ITEM_VARIANT_BLUE } from "../MenuItem/MenuItemVariant";
+import { MenuItemProps } from "./Menu";
+import {
+  MENU_PLACEMENT_BOTTOM_RIGHT,
+  MENU_PLACEMENT_LEFT_BOTTOM,
+  MENU_PLACEMENT_LEFT_HALF_TOP,
+  MENU_PLACEMENT_RIGHT_BOTTOM,
+  MENU_PLACEMENT_RIGHT_TOP
+} from "./MenuVariant";
+
+export interface MenuCollapseProps {
+  extraData?: any;
+  menu: MenuItemProps[];
+  setSelected?: (id: string) => void;
+  isDefaultOpen?: boolean;
+  placement?: string;
+  variant?: string;
+  menuItemVariant?: string;
+  children: ReactNode;
+  className?: string;
+  container?: HTMLDivElement | null;
+  setSelectedOption?: any;
+  classNameContentMenu?: string;
+  selectedOption?: string;
+}
+
+const MenuColapse = (props: MenuCollapseProps) => {
+  const {
+    menu,
+    placement = MENU_PLACEMENT_BOTTOM_RIGHT,
+    menuItemVariant = MENU_ITEM_VARIANT_BLUE,
+    variant,
+    children,
+    isDefaultOpen,
+    className,
+    container,
+    setSelectedOption,
+    classNameContentMenu,
+    selectedOption,
+    extraData
+  } = props;
+
+  const [isOpen, setIsOpen] = useState(isDefaultOpen);
+  const [openItems, setOpenItems] = useState<{ [key: string]: boolean }>({});
+
+  useEffect(() => {
+    const initialOpenItems = menu.reduce((acc, item) => {
+      if (item.type === "collapse") {
+        acc[item.id] = true;
+      }
+      return acc;
+    }, {} as { [key: string]: boolean });
+
+    setOpenItems(initialOpenItems);
+  }, [menu]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (menuContainerRef.current && !menuContainerRef.current?.contains(event.target as Node)) {
+        setIsOpen(false);
+      }
+    };
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, []);
```

Comment:
don't we have it as a reusable hook? if not, can you create it?

Reply:
done